### PR TITLE
Remove arg lookup map in CLIFactory

### PR DESCRIPTION
**Pull request by mik-laj**
We don't need this map. It only makes it difficult to add new commands because Pycharm has no key autocomplete for the dictionary.

---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -1082,35 +1155,46 @@ def get_long_option(arg):
...
+        dag_parser = False
+
+        for arg in cls.sort_args(sub['args']):
+            if 'dag_id' in arg.flags and dag_parser:
+                continue
+            kwargs = {
+                k: v for k, v in vars(arg).items() if k != 'flags' and v
+            }
+            sub_proc.add_argument(*arg.flags, **kwargs)
+        sub_proc.set_defaults(func=sub['func'])
```
Comment (turbaszek) [inline airflow/bin/cli.py]: What is the purpose of `dag_parser`? If I'm not mistaken it's always equal to `False` so the first condition from the loop is useless.

File: airflow/bin/cli.py [hunk 2]
```diff
@@ -1082,35 +1155,46 @@ def get_long_option(arg):
...
-                arg = cls.args[arg]
-                kwargs = {
-                    f: v
-                    for f, v in vars(arg).items() if f != 'flags' and v}
-                sub_proc.add_argument(*arg.flags, **kwargs)
-            sub_proc.set_defaults(func=sub['func'])
+            raise AirflowException("Invalid command definition.")
+
+    @classmethod
+    def _add_action_command(cls, sub, sub_proc):
```
Comment (turbaszek) [inline airflow/bin/cli.py:1178]: Would you mind adding type hints?

File: airflow/bin/cli.py [hunk 3]
```diff
@@ -1082,35 +1155,46 @@ def get_long_option(arg):
...
+            if 'dag_id' in arg.flags and dag_parser:
+                continue
+            kwargs = {
+                k: v for k, v in vars(arg).items() if k != 'flags' and v
+            }
+            sub_proc.add_argument(*arg.flags, **kwargs)
+        sub_proc.set_defaults(func=sub['func'])
+
+    @classmethod
+    def _add_subcommaand(cls, sub, sub_proc):
```
Comment (turbaszek) [inline airflow/bin/cli.py]:
```suggestion
    def _add_subcommand(cls, sub, sub_proc):
```

**Review by turbaszek — COMMENTED** [review]
I like the direction of this change!

File: airflow/bin/cli.py (hunk 2 above)
Comment (mik-laj) [reply airflow/bin/cli.py:1178]: I will gladly do it, but in a separate PR. In the next PR I want to replace dictionaries with named tuples. This is a good time to introduce type hints.

File: airflow/bin/cli.py (hunk 1 above)
Comment (mik-laj) [reply airflow/bin/cli.py]: This is part of the broken DAG.cli(). I hope this will be fixed sometime, so I did not delete this code.

Comment (mik-laj) [reply airflow/bin/cli.py]: Here is PR about DAG.cli() https://github.com/apache/airflow/pull/6875

File: airflow/bin/cli.py (hunk 3 above)
Comment (mik-laj) [reply airflow/bin/cli.py]: Fixed. Thanks.

File: airflow/bin/cli.py (hunk 1 above)
Comment (turbaszek) [reply airflow/bin/cli.py]: I think that we should not keep code that has no functionality. Once it will be restore we should restore the code.

Comment (mik-laj) [reply airflow/bin/cli.py]: I deleted this code.

File: airflow/bin/cli.py (hunk 3 above)
Comment (mik-laj) [reply airflow/bin/cli.py]: I came up with a better name  - `_add_group_command`

**Review by turbaszek — APPROVED** [review]

**Pull request merged**